### ISSUE TITLE
Git for recipies

### ISSUE THREAD

**Issue by Thrilleratplay** [open-source-ideas/open-source-ideas#104]
## Project description

An open source version of [cinc.kitchen](cinc.kitchen) where recipes are kept in a "git" repo in markdown and can be forked/started.  The "killer app" feature would be a way compare different forked  recipes with a rating (more healthy, more savory, less salty, ingredient substitution, etc)

This would probably pair nice with #103 

## Relevant Technology

Git like version control and any web technology with an interface to make edits and merges easier.

### Complexity

- [ ] Beginner - This project requires no or little prior knowledge of the technolog(y|ies) specified to contribute to the project
- [X ] Intermediate - The user should have some prior knowledge of the technolog(y|ies) to the point where they know how to use it, but not necessarily all the nooks and crannies of the technology
- [ ] Advanced - The project requires the user to have a good understanding of all components of the project to contribute

### Required time (ETA)

- [ ] Little work - A couple of days
- [X ] Medium work - A week or two
- [ ] Much work - The project will take more than a couple of weeks and serious planning is required

**Comment by KOLANICH** [comment]
>in markdown

This is unstructured informatikn. I guess writing recepies in a more formal language (I guess something around json) would be better.

**Comment by Spiffyk** [comment]
Hmm, what about XML for structure, Markdown for content? Something like:

```xml
<instructions>
    * some first step
    * some **second** step
    * yay, _Markdown_
</instructions>
```

**Comment by KOLANICH** [comment]
No. I meant something like
```javascript
"equipment":{
 "stove_for_baking":{"type":"stove", ....},
 "mixer":{"type":"mixer", ....}
},
"ingredients":{
"flour":{"what":"flour"},
"water":{"what": "water"},
},
"tasks":{
"task 1":
[
{op:"store", "mixer":[
{"what": "flour", "quantity":"1 glass"},
{"what": "water", "quantity":"1 glass"},
]},
{"op": "equipment", "mixer":{"op":"rotate", "time": ...., "rpm":"...."}
],
"task 2":[
{"op": "equpment", "stove_for_baking":{"op":"preheat", "temp":200}}
]

}
```

Must contain enough info to automate this fully, assumming availability of smart equipment and robots with hands.

**Comment by Thrilleratplay** [comment]
@KOLANICH You're right.  I was thinking markdown because written recipes are simple text list but to do comparisons or even metric to imperial unit conversations it would need to be structured.

**Comment by karuppiah7890** [comment]
Wow. Nice idea 😁  And about the markdown/ xml/ json discussion, we could consider yaml too. And if readability is a concern, we can choose markdown if UI is gonna be a file, so that it's human readable, and for processing we could use structured format. Then you would need some converters to convert from json/xml/yaml to markdown and show appropriate info, or we could just use front end Js and format things based on the data we get. Also, I have read in a few places that json serialization / deserialization is faster compared to xml, so, gotta consider that. Not sure about yaml

**Comment by KOLANICH** [comment]
>json/xml/yaml to markdown

It]s better to convert to dot and then render with graphviz

**Comment by jrickman** [comment]
There's an existing XML DTD for recipes called RecipeML http://www.formatdata.com/recipeml/ you might want to check out.

**Comment by KOLANICH** [comment]
Unfortunately it is free-form text.

**Comment by jueast08** [comment]
If its XML we are talking about, we can always redefine a structure with XMLSchema (though I'm a fan of json)

**Comment by Clschurr77** [comment]
Sent from Yahoo Mail for iPad


On Wednesday, August 15, 2018, 3:35 AM, KOLANICH <notifications@github.com> wrote:



in markdown


This is unstructured informatikn. I guess writing recepies in a more formal language (I guess something around json) would be better.

**Comment by Thrilleratplay** [comment]
Much of this functionality seems to be covered in [qgriffith/OpenEats](https://github.com/qgriffith/OpenEats)
